src/room.js

```diff
@@ -0,0 +1,37 @@
+class Room {
+
+  constructor (roomname) {
+    this.roomname = roomname;
+    this.endpoints = {};
+  }
+  addEndpoint (endpointId) {
+    this.endpoints[endpointId] = {
+      name: '',
+      permissions: 'CHAT',
+      commsid: ''
```

Comment:
Rather than initialising endpoints with some empty strings, we could either:
1. have a few more methods:
+ addUsername
+ addPermissions
+ addCommsId

2. change the "update" methods so that they can also be used for adding new username/permissions/commsId. For example:
```javascript
updateUsername (endpointId, username) {
  this.endpoints[endpointId][username] = username;
}
```